yt_dlp/extractor/googledrive.py

```diff
@@ -66,14 +66,106 @@ class GoogleDriveIE(InfoExtractor):
         '22': 'mp4',
         '34': 'flv',
         '35': 'flv',
+        # itag 36 videos are either 320x180 (BaW_jenozKc) or 320x240 (__2ABJjxzNo), abr varies as well
         '36': '3gp',
         '37': 'mp4',
         '38': 'mp4',
         '43': 'webm',
         '44': 'webm',
         '45': 'webm',
         '46': 'webm',
+        '50': 'mp3',
```

Comment:
Are you sure that `50` == `mp3` ? In the linked issue, two users said that `50` was `m4a` (and that `140` was `mp3`, but that seems wrong).

Can you add tests with example URLs for formats 50 and 140?

Reply:
My initial issue was actually with a mp3, adding 50 allows me to download, removing causes an error. I don't thing 140 is mp3, since it is already in YouTube.

Reply:
mp3 was the format I was initially trying to download, adding 50 solved the issue for me, removing it prevents me to download the content.
No idea for 140, but it is unlikely to be different from the YouTube extractor.

Reply:
It does make more sense to me that 50 would be mp3, and there wasn't enough log evidence in the issue to contradict that.

The PR just needs tests added; and make sure they do not violate copyrights

Reply:
For the test this is tricky as it doesn't work with some files only.

Even if I upload a mp3 on GDrive, I am able to download it without the patch. I would need a known failure, that is also royalty free.

Reply:
So, unpatched system-wide version with my test file:

```
/usr/bin/yt-dlp --version 
2024.04.09
```

```
/usr/bin/yt-dlp "https://drive.usercontent.google.com/download?id=1IP0o8dHcQrIHGgVyp0Ofvx2cGfLzyO1x"
[GoogleDrive] Extracting URL: https://drive.usercontent.google.com/download?id=1IP0o8dHcQrIHGgVyp0Ofvx2cGfLzyO1x
[GoogleDrive] 1IP0o8dHcQrIHGgVyp0Ofvx2cGfLzyO1x: Downloading video webpage
ERROR: 1IP0o8dHcQrIHGgVyp0Ofvx2cGfLzyO1x: An extractor error has occurred. (caused by KeyError('50')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
```

Local patched version with the test file:

```
yt-dlp "https://drive.usercontent.google.com/download?id=1IP0o8dHcQrIHGgVyp0Ofvx2cGfLzyO1x"
[GoogleDrive] Extracting URL: https://drive.usercontent.google.com/download?id=1IP0o8dHcQrIHGgVyp0Ofvx2cGfLzyO1x
[GoogleDrive] 1IP0o8dHcQrIHGgVyp0Ofvx2cGfLzyO1x: Downloading video webpage
[GoogleDrive] 1IP0o8dHcQrIHGgVyp0Ofvx2cGfLzyO1x: Requesting source file
[info] 1IP0o8dHcQrIHGgVyp0Ofvx2cGfLzyO1x: Downloading 1 format(s): source
[download] Resuming download at byte 523264
[download] Destination: My Buddy - Henry Burr - Gus Kahn - Walter Donaldson.mp3 [1IP0o8dHcQrIHGgVyp0Ofvx2cGfLzyO1x].mp3
[download] 100% of    3.30MiB in 00:00:02 at 1.60MiB/s
```

So this file is okay for a test.

The file is sourced from https://archive.org/details/78_my-buddy_henry-burr-gus-kahn-walter-donaldson_gbia0015627a/My+Buddy+-+Henry+Burr+-+Gus+Kahn+-+Walter+Donaldson.flac
from 1922, royalty free now.